app/controllers/orders_controller.rb

```diff
@@ -30,20 +30,23 @@ def show
   end
 
   def update
-    @order = Order.find(params[:id])
-    if params[:drone]
+    if current_user.store
+      store = Store.find(current_user.store.id)
+      @order = store.orders.find(params[:id])
       @order.update(drone: params[:drone].to_i)
-      flash[:success] = "Order ##{@order.id} is #{@order.drone}."
-      if current_user.store
-        Drone.request_drone()
-        redirect_to admin_order_path(@order.id)
-      else
+      Drone.request_drone() if params[:drone].to_i == 3
```

Comment:
Maybe we should extract the conditional to a helper method that's more explicit.